### PR TITLE
Source Github: deprecate "page size" input parameter

### LINKED ISSUE #10847: Source Github: deprecate "page size" input parameter
## Tell us about the problem you're trying to solve
Like mentioned in the [UX handbook](https://docs.airbyte.io/connector-development/ux-handbook) this parameter is asking the user for input about an implementation detail. The user has no way of knowing what the right value for this parameter should be. 
Screen Shot 2022-03-04 at 12 15 06 AM

## Describe the solution you’d like
Remove that option and use a sane value by default.

### LINKED ISSUE #25025: Repository field on GitHub connector doesn't accept full URL
Need to improve setup success rate on GitHub connector
# problem
We watched fullstory sessions after noticing a high give up rate. In the repository field, most people input the full URL e.g “https://github.com/openai/gpt-3.git” and even though the error message said specifically what the correct format is, they didn’t know what to fix. We require just the repository name, not the full URL, so it fails to setup. 

 @lazebnyi how do you recommend we fix?

### PR DESCRIPTION
**Pull request by grubberr**
## What

Issue: https://github.com/airbytehq/airbyte/issues/10847

- Remove "page_size_for_large_streams" from `spec.json`, but still use it for current setups

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-github/source_github/source.py [hunk 1]
```diff
@@ -133,6 +133,7 @@ def _get_org_repositories(config: Mapping[str, Any], authenticator: MultipleToke
             stream = RepositoryStats(
                 authenticator=authenticator,
                 repositories=unchecked_repos,
+                # This parameter is deprecated and in future will be used sane default, page_size: 10
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-github/source_github/source.py:136]: Can we add sone unit test for cover backward compatibility.

**Review by lazebnyi — CHANGES_REQUESTED** [review]
LGTM! But need add unit tests.

File: airbyte-integrations/connectors/source-github/source_github/source.py (hunk 1 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-github/source_github/source.py:136]: In really I have only removed OPTIONAL param "page_size_for_large_streams" from `spec.json` and nothing more for now.

Hm I am stuck finding how good way to test this case.

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-github/source_github/source.py:136]: Attempt to read data using a custom page size or default and then compare the resulting quantity of records with the expected amount.

**Review by lazebnyi — APPROVED** [review]
Approved to unblock. But still need add unit tests.

**Pull request merged**